### PR TITLE
A collection of small fixes

### PR DESCRIPTION
**Pull request by tiennou**
This is a handful of small test-harness tweaks I collected while trying to have Xcode run the test cases (for the record, you need to set your PATH env-var correctly so Xcode finds `docker`, for each separate test case you need to debug).

The second commit moves our headers first in the list, so Xcode wouldn't try to use the ones from "system" locations (ie. Homebrew's `/usr/local`).

The last one hides OpenSSL's AES-CTR thingy inside its own initialization so it doesn't force other backends to `NOP` it.

### CODE REVIEW

**Comment by tiennou** [comment]
Rebased, anyone ?

**Comment by willco007** [comment]
The AppVeyor builds are failing, could you look into why? Other than that, the actual refactoring looks good to me.

**Comment by tiennou** [comment]
Should be fixed. I fixup-ed some defines for missing headers on MSVC, and it's starting to come back green 🤞.

**Comment by willco007** [comment]
Taking more of a look at this, some of the formatting needs to be updated/reverted to match the formatting guidelines. For example, no spaces after if/for and before ( and the else bracket formatting. That will cut down on the churn as well. After that's fixed, I'm ready to take these changes.

File: src/openssl.h [hunk 1]
```diff
@@ -280,18 +280,7 @@ int _libssh2_md5_init(libssh2_md5_ctx *ctx);
...
-  ENGINE_register_all_complete()
-#else
-#define libssh2_crypto_init() \
-  OpenSSL_add_all_algorithms(); \
-  OpenSSL_add_all_ciphers(); \
-  ENGINE_load_builtin_engines(); \
-  ENGINE_register_all_complete()
-#endif
-
+#define libssh2_crypto_init() _libssh2_openssl_crypto_init()
```
Comment (tiennou) [inline src/openssl.h]: Needs a typedef.

File: tests/openssh_fixture.c [hunk 2]
```diff
@@ -57,87 +57,98 @@
...
-    char command_buf[BUFSIZ];
-    int rc = snprintf(command_buf, sizeof(command_buf), "docker stop %s",
-                      container_id);
-    if(rc > -1 && rc < BUFSIZ) {
-        return run_command(command_buf, NULL);
-    }
-    else {
-        return rc;
-    }
+	return run_command(NULL, "docker stop %s", container_id);
```
Comment (tiennou) [inline tests/openssh_fixture.c]: Indent

File: tests/session_fixture.c [hunk 3]
```diff
@@ -50,58 +54,80 @@
...
+{
+	char *wd = getenv("FIXTURE_WORKDIR");
+#ifdef FIXTURE_WORKDIR
+	if(!wd) {
+		wd = FIXTURE_WORKDIR;
+	}
+#endif
+	if(!wd) {
+#ifdef WIN32
+        char wd_buf[_MAX_PATH];
```
Comment (tiennou) [inline tests/session_fixture.c]: Indent

File: tests/session_fixture.c [hunk 4]
```diff
@@ -50,58 +54,80 @@
...
-                return NULL;
-            }
-        }
-        else {
-            fprintf(stderr, "libssh2_init failed (%d)\n", rc);
-            return NULL;
-        }
+    int rc;
+
+	setup_fixture_workdir();
```
Comment (tiennou) [inline tests/session_fixture.c]: Indent

**Comment by willco007** [comment]
Oh that's my bad; I was diff'ing against an old branch accidentally so it was showing incorrect changes. Your changes looks good.

**Comment by tiennou** [comment]
Should be ready now, I took care of the issues I saw. Glad the style is OK 😉.

**Pull request merged**